docs/spark_expressions_support.md

```diff
@@ -0,0 +1,475 @@
+<!---
+  Licensed to the Apache Software Foundation (ASF) under one
+  or more contributor license agreements.  See the NOTICE file
+  distributed with this work for additional information
+  regarding copyright ownership.  The ASF licenses this file
+  to you under the Apache License, Version 2.0 (the
+  "License"); you may not use this file except in compliance
+  with the License.  You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing,
+  software distributed under the License is distributed on an
+  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+  KIND, either express or implied.  See the License for the
+  specific language governing permissions and limitations
+  under the License.
+-->
+
+# Supported Spark Expressions
+
+### agg_funcs
+ - [x] any
+ - [x] any_value
+ - [ ] approx_count_distinct
+ - [ ] approx_percentile
+ - [ ] array_agg
+ - [x] avg
+ - [x] bit_and
+ - [x] bit_or
+ - [x] bit_xor
+ - [x] bool_and
+ - [x] bool_or
+ - [ ] collect_list
+ - [ ] collect_set
+ - [ ] corr
+ - [x] count
+ - [x] count_if
+ - [ ] count_min_sketch
+ - [x] covar_pop
+ - [x] covar_samp
+ - [x] every
+ - [x] first
+ - [x] first_value
+ - [ ] grouping
+ - [ ] grouping_id
+ - [ ] histogram_numeric
+ - [ ] kurtosis
+ - [x] last
+ - [x] last_value
+ - [x] max
+ - [ ] max_by
+ - [x] mean
+ - [ ] median
+ - [x] min
+ - [ ] min_by
+ - [ ] mode
+ - [ ] percentile
+ - [ ] percentile_approx
+ - [x] regr_avgx
+ - [x] regr_avgy
+ - [x] regr_count
```

Comment:
I don't think that we support these expressions

Reply:
The test is exactly for year, but if only YEAR supported, what is supposed to show to the user? Not supported?

Reply:
```
test("regr_avgx") {
    Seq(false, true).foreach { dictionary =>
      withSQLConf(
        "parquet.enable.dictionary" -> dictionary.toString,
        "spark.comet.exec.shuffle.enabled" -> "true",
        CometConf.COMET_ENABLED.key -> "true",
        CometConf.COMET_EXEC_ENABLED.key -> "true",
        CometConf.COMET_SHUFFLE_ENFORCE_MODE_ENABLED.key -> "true",
        CometConf.COMET_EXEC_ALL_OPERATOR_ENABLED.key -> "true",
      ) {
        val table = "test"
        withTable(table) {
          sql(s"create table $table(a int, b int) using parquet")
          sql(s"insert into $table VALUES (1, 2), (2, 2), (2, 3), (2, 4)")
          checkSparkAnswerAndOperator(s"SELECT regr_avgx(a, b) FROM $table")
        }
      }
    }
  }
```

regr_avgx test passes

Reply:
regr_avgx supported by DF
```
> SELECT regr_avgx(1, 2);
+------------------------------+
| REGR_AVGX(Int64(1),Int64(2)) |
+------------------------------+
| 2.0                          |
+------------------------------+
```
so I think all is fair here